android/src/ti/intercom/TitaniumIntercomModule.java

```
@@ -54,7 +56,13 @@ public static void onAppCreate(TiApplication app) {
 		apiKey = app.getAppProperties().getString("intercom-android-api-key", null);
 		appId = app.getAppProperties().getString("intercom-app-id", null);
 
-		initialize();
+		// Defer the initialization on Android 12 and 13 to avoid ANRs as suggested by the Intercom Support team.
+		// It would further require that the Intercom.initialize() is called before any other Intercom API is used.
+		if (Build.VERSION.SDK_INT >= 31 && Build.VERSION.SDK_INT <= 33) {
+			Intercom.registerForLaterInitialisation(app);
+		} else {
+			initialize();
```

What does that mean for the app-side code? As we do not initialize Android on open anymore, doesn't this case issues for this Android version?

---

- Android 12/13 will be initialised when the main landing page is opened, rest will be done here.
- The `initialize()` call silently fails if the Intercom is already initialised on any version, so it's all safe.